plugins/CorePluginsAdmin/Controller.php

```diff
@@ -425,9 +425,22 @@ public function safemode($lastError = array())
 
     public function activate($redirectAfter = true)
     {
-        $pluginName = $this->initPluginModification(static::ACTIVATE_NONCE);
         $this->dieIfPluginsAdminIsDisabled();
 
+        $params = [
+            'module' => 'CorePluginsAdmin',
+            'action' => 'activate',
+            'pluginName' => Common::getRequestVar('pluginName'),
+            'nonce' => Common::getRequestVar('nonce'),
+            'redirectTo' => Common::getRequestVar('redirectTo'),
```

Comment:
That actually seems not to work as expected. `redirectTo` might be set to `referer`. In this case the method redirects to the referrer sent with the request. As the password verify now adds another redirect in between, the referrer isn't the original anymore, causing a loop on the confirmation page.

Reply:
:+1: should be fixed